wagtail_airtable/management/commands/import_airtable.py

```diff
@@ -116,6 +118,42 @@ def convert_mapped_fields(self, record_fields_dict, mapped_fields_dict) -> dict:
         }
         return mapped_fields_dict
 
+    def update_model_m2m_fields(self, instance, field_name, value) -> None:
+        m2m_field = getattr(instance, field_name)
+        for m2m_value in value:
+            m2m_field.add(m2m_value)
+
+    def handle_model_data_fields(self, model, data: dict, instance: None):
```

Comment:
Thanks for consolidating! This is a bit of a nitpick, but I think this function possibly does a few too many things conditionally right now, which is a little confusing - if it's got an instance, it sets variables + updates the m2m fields, and if it doesn't, then it returns dictionaries of variables and m2m fields instead.

Suggestion: could we either make the loop over fields a generator, yielding for example `field_name, value, is_m2m` at each stage, and have both "separate regular and m2m fields" and "set all fields" use this same generator to loop over these fields? Or alternatively (at the cost of an extra loop, but who's counting?) have this function just separate non-m2m and m2m fields, returning them, and have other functions take care of the setting?

To me, this just feels a little too conditional to work well as a function on its own, since it needs a bit of thinking through to work out what's going to happen

Reply:
I agree the function needs to be simplified a little. I created a generator, yielding as you suggested and used it when creating, updating objects. let me know if that's what you had in mind